### PR TITLE
Configuring for iOS Auto-Reconnect

### PR DESCRIPTION
**Pull request by oleander**
The configuration options mentioned in the README work reliably for iOS devices. Unfortunately, I don't have any sources to back up my statements, but it works for all my iOS 16 & 17 devices against ESP32 and ESP32C3.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,3 +18,30 @@ CONFIG_BT_NIMBLE_ENABLED=y
...
+In your Rust implementation, set the security options as follows:
+
+```rust
+device
+  .security()
+  .set_auth(AuthReq::Bond) // or .set_auth(AuthReq::Bond | AuthReq:Mitm)
+  .set_io_cap(SecurityIOCap::NoInputNoOutput) // Any option here works
+```
+
+Here, `.set_auth(AuthReq::Bond)` is used to enable bonding, which stores security keys for future reconnections. Note that the use of `.set_auth(AuthReq::Sc)` (Secure Connections) prevents iOS from reconnecting.
```
Comment (taks) [inline README.md:47]: In isssue #24, we concluded that if `sm_our_key_dist` and `sm_their_key_dist` are set, it is possible to reconnect even if Secure Connections is used.
These settings can be set by using the [`resolve_rpa()`](https://taks.github.io/esp32-nimble/esp32_nimble/struct.BLESecurity.html#method.resolve_rpa) function.

If it is possible to reconnect using this function in your environment, please modify this comment.

**Comment by oleander** [comment]
I did some testing using `resolve_rpa`. I must enable `CONFIG_BT_NIMBLE_NVS_PERSIST`; otherwise, I cannot reconnect, regardless of the other settings. It doesn't matter if I use `set_own_addr_type_to_non_resolvable_random` before connecting to an iOS device.

What's new for me is that I can now use `AuthReq::all()` combined with `CONFIG_BT_NIMBLE_NVS_PERSIST` to allow my iOS devices to reconnect. That just started working as before I could not use `AuthReq::Sc`.

Do we need more devices to test against, or am I doing something wrong?

**Comment by oleander** [comment]
Is there anything else I can do to get this merged?

**Comment by taks** [comment]
I would like to change the last section to use `resolve_rpa()`.
What do you think?

**Comment by oleander** [comment]
I added `resolve_rpa` and cleaned up the language a bit.

**Comment by taks** [comment]
```
- **Caution**: Avoid `.set_auth(AuthReq::Sc)` (Secure Connections), as it can interfere with the ability of iOS devices to reconnect.
```
Regarding the above statement,
I'm guessing that using `.set_auth(AuthReq::Sc)` is not a problem if `.resolve_rpa()` is used.
Is it wrong?
Sorry, I don't have iOS device, so I can't test it.

**Comment by oleander** [comment]
Let me test later tonight

**Comment by oleander** [comment]
I couldn't get my failing code to … fail using `Auth::Sc`, so this took a while to test. I, therefore, removed the warning about using `Auth::Sc` with iOS.

**Comment by taks** [comment]
Thank you for testing.

**Pull request merged**